beater/journalbeat.go

```diff
@@ -306,6 +322,60 @@ func (jb *Journalbeat) Run(b *beat.Beat) error {
 
 	jb.client = b.Publisher.Connect()
 
+	var err error
+	numLogstashInstances, err := b.Config.Output["logstash"].CountField("hosts")
+	if err != nil {
+		logp.Err("Invalid configuration for sending contents to logstash")
+		os.Exit(101)
+	}
+
+	//parse all the different endpoints read-only iteration.
+	var endpoints []string
+	for i := 0; i < numLogstashInstances; i++ {
+		endpoint, err := b.Config.Output["logstash"].String("hosts", i)
+		fmt.Println("Retrieved endpoints is" + endpoint)
+		if err != nil {
+			logp.Err("Invalid configuration failed to lookup endpoint for")
+			os.Exit(102)
+		}
+		//check if the logstash endpoint is reachable
+		conn, err := net.Dial("tcp", endpoint)
+		defer conn.Close()
+		if err != nil {
+			logp.Err("Error encountered connecting to logstash endpoint: %s", err)
```

Comment:
This means that whatever is not available at startup time, will never be used for that host session lifetime.
As discussed, let's remove this check, and we'll retry forever to those instances. That's still a problem in runtime, if one instance was present during initialization but stops working, no job assigned to that instance will progress.

Reply:
Since our logstash instances are being managed by aurora this should not happen. I would suggest getting rid of the rack constraints in aurora for the logstash aggregators since I found in some of the datacenter 2/5 instances were not running because of this. If all the logstash aggregators are not available for 2-3 minutes it will not be end of the world  since we buffer logs locally on the hosts. 

The rack constraints/anycast ip address seems to be a workaround because we did not have proper sharding in place and were using anycast as a rough approximation :)